### PR TITLE
Datasource for Grafana logging platform

### PR DESCRIPTION
**Pull request by davkal**
- new builtin datasource plugin "Logging" (likely going to be renamed)
- plugin implements no panel ctrls yet, only ships datasource
- new models for logging data as first class citizen (aside from table
  and time_series model)
- Logs as new view for Explore
- colored log levels, highlighted search results
- JSON view for development

Testable only against existing logish deployment.
Then test with queries like `{job="..."} regexp`.

Next steps (not this PR):

- use own query field when #12643 is merged
- show stream labels and add interaction
- show more metadata about streams
- better empty state
- logs panel for dashboards with query ctrl

### CODE REVIEW

File: public/app/core/logs_model.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  crit = 'crit',
+  warn = 'warn',
+  err = 'error',
+  error = 'error',
+  info = 'info',
+  debug = 'debug',
+  trace = 'trace',
+}
+
+export interface LogSearchMatch {
```
Comment (torkelo) [inline public/app/core/logs_model.ts]: thinking of having these pure types definitions in app/types, not sure if that is a good idea or not.

File: public/app/plugins/datasource/logging/result_transformer.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    matches.push({
+      text: match[0],
+      start: match.index,
+      length: match[0].length,
+    });
+  }
+  return matches;
+}
+
+export function processEntry(entry: { line: string; timestamp: string }, stream): LogRow {
```
Comment (torkelo) [inline public/app/plugins/datasource/logging/result_transformer.ts]: this looks like good function to to test as well, although I see now that most is tested via tests for getSearchMatches & getLogLevel so not super critical.

**Review by torkelo — APPROVED** [review]
The code looks good, let me know if you need feedback on UX side (only looked at code)

File: public/app/plugins/datasource/logging/result_transformer.ts (hunk 2 above)
Comment (davkal) [reply public/app/plugins/datasource/logging/result_transformer.ts]: Will do in a later PR.

File: public/app/core/logs_model.ts (hunk 1 above)
Comment (davkal) [reply public/app/core/logs_model.ts]: Yeah, still *exploring* this.

**Pull request merged**